### DISCUSSION TITLE
Field nullability

### DISCUSSION THREAD

**Discussion by NN---** [dotnet/csharplang#4186] [Q&A]
How can I specify that field must have non null value when assigned but until that it doesn't hold any value ?
It would be nice to have some attribute for this behavior:

```cs
#nullable enable

class A
{
  [MemberNotNull] // Proposal: null value cannot be assigned
  string? something; // null now

 
  public string F(string x) => something ??= x; // OK

  public string G(string? y) => something ??= y; // Error, null value cannot be assigned.
}
```

**Comment by Joe4evr** [comment]
```cs
[DisallowNull]
string? something;
```

**Reply by canton7** [reply]
[See this on SharpLab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBMQGoA+ABATARgFgAobABgAJt8A6AEQEsIBzAOwHtYGBjKGgYXYo4AQVYQANgE8oDKAG4SAYlYBXCRIjAJcCnHHa4JEtgDMVXBX4UA3iQoOKAbUZRJE9gHcAcuokBde0dqMgB+Cih2MDgYAAsGVmZFYkcKIIczKgAWCgBZAAoASlt01Kp8AE58yOi4hOYaABl9ZjjC+QoAek6KAB4AWn6KTwgEVnrS1JqY+MSKAF4KNQ0O7r7B4dHxxMnHabq5xYAiADN2diPksuDK6qiZ+qaWttWegaGOTbGJlMcAXxIfyAA=).

See [this article](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/attributes/nullable-analysis) for a good run-down on (most of) the NRT-related attributes.

**Reply by NN---** [reply]
@Joe4evr Thanks.
It is discovered that ReSharper showed me error for such code.
But in fact there is no compilation issue.
Opened an issue for ReSharper.

**Comment by NN---** [comment]
@Joe4evr It turns out that this works unexpectedly :) 
https://github.com/dotnet/roslyn/issues/49903
https://github.com/dotnet/roslyn/pull/49911
